src/django_components/component_media.py

```diff
@@ -398,7 +399,11 @@ def check_pair_empty(inline_attr: str, file_attr: str) -> bool:
     return None
 
 
-media_cache: Dict[Type["Component"], MediaCls] = {}
+# NOTE: We use weakref to avoid issues with lingering references.
+if sys.version_info >= (3, 9):
```

Comment:
Updated so that `media_cache` does not hold a strong reference to the Component classes